### PR TITLE
Map generation limit: Make per-world 

### PR DESCRIPTION
**Pull request by paramat**
The setting limits map generation but affects nothing else.
Add 'mapgen_limit' to global mapgen parameters.
Move 'blockpos_over_mapgen_limit()' to the only place it is called
from: map.cpp.
Allow teleportation to any part of the world even if over the set
mapgen limit.
Simplify the reading of this limit in mgvalleys.
Remove the 'map_generation_limit' setting.
//////////////////////////////////////////////

Addresses #1608 
This is a new version of #4338 i have copied some of the code by SmallJoker.

### CODE REVIEW

**Comment by rubenwardy** [comment]
I think that this would be better implemented by making per-world settings, personally

**Comment by paramat** [comment]
Mapgen limit is a setting but should not have been, it is a fundamental mapgen parameter like water_level. For some settings i would agree with you but this one i feel belongs grouped with the other basic mapgen paramters.

**Comment by paramat** [comment]
In testing this works.
I am not happy with how the mapgen limit setting is grabbed by passing `settings_mgr.mapgen_params` from the call, how can we grab the value directly in `mapblock.h`?

**Comment by paramat** [comment]
Updated, need to test.

**Comment by paramat** [comment]
Tested, adds the new parameter to exisiing worlds on update.
I'll add docs once this gets some approval.

File: src/mapgen.h [hunk 1]
```diff
@@ -124,6 +124,7 @@ struct MapgenParams {
 	s16 chunksize;
 	u64 seed;
 	s16 water_level;
+	s16 mapgen_limit;
```
Comment (SmallJoker) [inline src/mapgen.h]: `u16`

**Comment by paramat** [comment]
Added docs and re-added new setting to defaultsettings.cpp (oops).

**Comment by paramat** [comment]
Retested and fine.

**Comment by paramat** [comment]
ec0c4d33db9e0b7b3e541757e34c04c08c3b48c9

**Pull request closed without merging**